src/components/app/app.js

```diff
@@ -1,6 +1,23 @@
 import angular from "angular";
 import template from "./app.html";
 
-export const appModule = angular.module("appModule", []).component("app", {
-  template
-}).name;
+import ngMaterial from "angular-material";
+
+import { transactionDialogModule } from "../transaction-dialog/transaction-dialog";
+
+export class AppController {
+  constructor() {
+    this.transactionDialogOpen;
+  }
+
+  onAddTransactionClick(event) {
+    this.transactionDialogOpen = { event: event };
```

Comment:
what this line does?

Reply:
every press on button sends message to dialog to open and sends events, from where should dialog be shown